components/style/values/specified/mod.rs

```diff
@@ -339,47 +361,60 @@ impl ToComputedValue for Angle {
     }
 
     fn from_computed_value(computed: &Self::ComputedValue) -> Self {
-        Angle {
-            radians: computed.radians(),
-            was_calc: false,
-        }
+        Angle::from_radians(computed.radians())
     }
 }
 
 impl Angle {
-    #[inline]
-    #[allow(missing_docs)]
-    pub fn radians(self) -> f32 {
-        self.radians
+    /// Returns an angle with the given value in degrees.
+    pub fn from_degrees(value: CSSFloat) -> Self {
+        Angle { value: value, unit: AngleUnit::Degree, was_calc: false }
     }
-
-    /// Returns an angle value that represents zero radians.
-    pub fn zero() -> Self {
-        Self::from_radians(0.0)
+    /// Returns an angle with the given value in gradians.
+    pub fn from_gradians(value: CSSFloat) -> Self {
+        Angle { value: value, unit: AngleUnit::Gradian, was_calc: false }
+    }
+    /// Returns an angle with the given value in turns.
+    pub fn from_turns(value: CSSFloat) -> Self {
+        Angle { value: value, unit: AngleUnit::Turn, was_calc: false }
+    }
+    /// Returns an angle with the given value in radians.
+    pub fn from_radians(value: CSSFloat) -> Self {
+        Angle { value: value, unit: AngleUnit::Radian, was_calc: false }
     }
 
     #[inline]
     #[allow(missing_docs)]
-    pub fn from_radians(r: f32) -> Self {
-        Angle {
-            radians: r.min(f32::MAX).max(f32::MIN),
```

Comment:
Please ensure to keep the value finite, since there's other code that relies on this. Please adjust the constructors appropriately, and also the `radians` method.